packages/frontend/src/lib/notification/ContainerConnectionStatusInfo.svelte

```diff
@@ -0,0 +1,94 @@
+<script lang="ts">
+import Fa from 'svelte-fa';
+import { faCircleExclamation } from '@fortawesome/free-solid-svg-icons';
+import Button from '../button/Button.svelte';
+import { filesize } from 'filesize';
+import { studioClient } from '/@/utils/client';
+import type { ContainerConnectionInfo } from '@shared/src/models/IContainerConnectionInfo';
+
+export let connectionInfo: ContainerConnectionInfo;
+type BannerBackgroundColor = 'light' | 'dark';
+export let background: BannerBackgroundColor = 'light';
+
+let title: string | undefined = '';
+let description: string | undefined = '';
+let actionName: string | undefined = '';
+$: updateTitleDescription(connectionInfo);
+
+function updateTitleDescription(connectionInfo: ContainerConnectionInfo) {
+  if (connectionInfo.status === 'no-machine') {
+    title = 'No Podman machine is running';
+    description = 'Please start a Podman Machine before proceeding further.';
+    actionName = 'Start now';
+    return;
+  }
+
+  if (connectionInfo.status === 'not-enough-resources') {
+    title = 'Upgrade your Podman machine for best AI performance';
+
+    const hasEnoughCPU = connectionInfo.cpus >= connectionInfo.cpusExpected;
+    const hasEnoughMemory = connectionInfo.memoryIdle > connectionInfo.memoryExpected;
+
+    let machineCurrentStateDescription = '';
+    let machinePreferredStateDescription = '';
+    if (hasEnoughCPU) {
+      machineCurrentStateDescription += `${connectionInfo.cpus} vCPUs`;
+      machinePreferredStateDescription += `${connectionInfo.cpusExpected} vCPUs`;
+      if (!hasEnoughMemory) {
+        machineCurrentStateDescription += ` and ${filesize(connectionInfo.memoryIdle, { base: 2 })} of memory available`;
+        machinePreferredStateDescription += ` and ${filesize(connectionInfo.memoryExpected, { base: 2 })} of memory`;
+      }
+    } else {
+      machineCurrentStateDescription += `${filesize(connectionInfo.memoryIdle, { base: 2 })} of memory available`;
+      machinePreferredStateDescription += `${filesize(connectionInfo.memoryExpected, { base: 2 })} of memory`;
+    }
+
+    const machineName = `${connectionInfo.name.includes('Podman Machine') ? connectionInfo.name : `Podman Machine ${connectionInfo.name}`}`;
+    description = `Your ${machineName} has ${machineCurrentStateDescription}.`;
+
+    if (connectionInfo?.canEdit) {
```

Comment:
Would `upgradeRecommended` be more representative than `canEdit` ? I feel like canEdit would mean if we are able or not to edit the podman machine, rather that should be edit for better performance